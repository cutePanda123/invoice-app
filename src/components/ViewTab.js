import React from 'react';
import Ionicon from 'react-ionicons';
import PropTypes from 'prop-types';
import Constants from '../utilility';

const generateTabClassName = (curTabName, activeViewName) => {
    return (curTabName === activeViewName) ? "nav-link active" : "nav-link";
}

const ViewTab = ({activeViewName, onClickTab}) => {
    return (
        <ul className="nav nav-tabs nav-fill my-4">
            <li className="nav-item">
                <a 
                    className={generateTabClassName(Constants.LIST_VIEW_NAME, activeViewName)}
                    href="#"
                    onClick={(event) => {
                        event.preventDefault();
                        onClickTab(Constants.LIST_VIEW_NAME);
                    }}
                >
                    <Ionicon
                        className="rounded-circle mr-2"
                        fontSize="25px"
                        color={"#007bff"}
                        icon="ios-paper"
                    />
                    List View
                </a>
            </li>
            <li className="nav-item">
                <a
                    className={generateTabClassName(Constants.CHART_VIEW_NAME, activeViewName)}
                    href="#"
                    onClick={(event) => {
                        event.preventDefault();
                        onClickTab(Constants.CHART_VIEW_NAME);
                    }}
                >
                    <Ionicon
                        className="rounded-circle mr-2"
                        fontSize="25px"
                        color={"#007bff"}
                        icon="ios-pie"
                    />
                    Chart View
                </a>
            </li>
        </ul>
    );
}

ViewTab.propTypes = {
    activeViewName: PropTypes.string.isRequired,
    onClickTab: PropTypes.func.isRequired
}

export default ViewTab;