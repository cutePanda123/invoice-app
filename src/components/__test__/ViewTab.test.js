import React from 'react';
import { shallow } from 'enzyme';
import ViewTab from '../ViewTab';
import Utility from '../../utility';

const props = {
    activeViewName: Utility.LIST_VIEW_NAME,
    onClickTab: jest.fn()
}

let wrapper = null;

describe('ViewTab test', () => {
    beforeEach(() => {
        wrapper = shallow(<ViewTab {...props} />);
    });

    it('snapshot test', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correct number of tabs', () => {
        const tabs = wrapper.find('.nav-item');
        expect(tabs.length).toEqual(2);
    });

    it('should trigger correct callback on list view', () => {
        const tab1 = wrapper.find('.nav-item a').first();
        tab1.simulate('click', { preventDefault: () => {} });
        expect(props.onClickTab).toHaveBeenCalledWith(Utility.LIST_VIEW_NAME);
    });

    it('should trigger correct callback on chart view', () => {
        const tab = wrapper.find('.nav-item a').last();
        tab.simulate('click', { preventDefault: () => {} });
        expect(props.onClickTab).toHaveBeenCalledWith(Utility.CHART_VIEW_NAME);
    });
});