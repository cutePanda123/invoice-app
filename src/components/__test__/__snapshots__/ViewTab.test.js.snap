// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTab test snapshot test 1`] = `
<ul
  className="nav nav-tabs nav-fill my-4"
>
  <li
    className="nav-item"
  >
    <a
      className="nav-link active"
      href="#"
      onClick={[Function]}
    >
      <Ionicon
        beat={false}
        className="rounded-circle mr-2"
        color="#007bff"
        fontSize="25px"
        icon="ios-paper"
        rotate={false}
        shake={false}
        style={Object {}}
      />
      List View
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      <Ionicon
        beat={false}
        className="rounded-circle mr-2"
        color="#007bff"
        fontSize="25px"
        icon="ios-pie"
        rotate={false}
        shake={false}
        style={Object {}}
      />
      Chart View
    </a>
  </li>
</ul>
`;
